wallet_manager.go

```diff
@@ -635,16 +650,18 @@ func (wm *WalletManager) EnsureTxWithHooks(
 			} else {
 				// if it is mined, we don't need to do anything, just stop the loop and return
 				for txhash, status := range statuses {
-					if status.Status == "done" || status.Status == "reverted" {
+					switch status.Status {
```

Comment:
using switch case when there is only one case is weird. Also, this does not handle default case.
I think in this case, using "if" is more appropriate